lib/src/main/java/uk/gov/di/ipv/cri/address/library/service/AddressSessionService.java

```diff
@@ -138,7 +142,7 @@ public TokenRequest createTokenRequest(String requestBody)
         boolean invalidTokenRequest =
                 request.getQueryParameters()
                         .keySet()
-                        .containsAll(List.of(CODE, CLIENT_ID, REDIRECT_URI, GRANT_TYPE));
+                        .containsAll(Set.of(CODE, CLIENT_ID, REDIRECT_URI, GRANT_TYPE));
```

Comment:
the api gateway should be doing this validation - and I'm torn about whether we should chek it twice - not a blocker, more an FYI that we need a conversation about this

Reply:
Ok I understand, I have done this b'cos of Will Palmer, I think this issue came up in one of our team wide discussion and he seemed to suggest that have multiple validation doesn't hurt. I suppose if we to bypass api-gateway for any reason then it might be useful. Can we call api gateway endpoints directly in a statemachine?